lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -103,8 +105,15 @@ impl RegionInternal for MmapRegion {
         ]
         .iter()
         {
-            // eprintln!("setting r/w {:p}[{:x}]", *ptr, len);
-            unsafe { mprotect(*ptr, *len, ProtFlags::PROT_READ | ProtFlags::PROT_WRITE)? };
+            unsafe {
+                let res = mprotect(*ptr, *len, ProtFlags::PROT_READ | ProtFlags::PROT_WRITE);
+
+                // If the mprotect() call failed, manually return the slot.
+                if let Err(e) = res {
```

Comment:
Talking with @pchickey I am starting feel that perhaps a panic! would be good here.  Thinking through the ways that mprotect() could fail, I'm wondering how one might recover gracefully from that.

Reply:
I was wrong, an `expect` is the right call here, much as it already is when we drop the alloc and `mprotect` back to `PROT_NONE`

Reply:
Thanks for this additional comment.  Very helpful.